packages/core-components/src/components/DismissableBanner/DismissableBanner.tsx

```diff
@@ -94,7 +94,10 @@ export const DismissableBanner = (props: Props) => {
   }, [observedItems?.newValue]);
 
   const handleClick = () => {
-    notificationsStore.set('dismissedBanners', [...dismissedBanners, id]);
+    notificationsStore.set<string[]>(
```

Comment:
Isn't the problem here that we're not calling `.get` again to get the latest value?